charts/newrelic-infrastructure/templates/controlplane/daemonset.yaml

```diff
@@ -18,50 +18,63 @@ spec:
   {{- end }}
   selector:
     matchLabels:
-      {{- include "newrelic.selectorLabels" . | nindent 6 }}
+      {{- include "common.labels.selectorLabels" . | nindent 6 }}
       app.kubernetes.io/component: controlplane
   template:
     metadata:
       annotations:
-        checksum/nri-kubernetes: {{ include (print $.Template.BasePath "/controlplane/configmap.yaml") . | sha256sum }}
-        {{- if not (include "newrelic.licenseCustomSecretName" .) }}
-        checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
-        {{- end }}
-        checksum/agent-config: {{ include (print $.Template.BasePath "/agent-configmap.yaml") . | sha256sum }}
+        checksum/nri-kubernetes: {{ include (print $.Template.BasePath "/controlplane/scraper-configmap.yaml") . | sha256sum }}
+        checksum/agent-config: {{ include (print $.Template.BasePath "/controlplane/agent-configmap.yaml") . | sha256sum }}
+        checksum/license-secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
         {{- with .Values.podAnnotations }}
         {{- toYaml . | nindent 8 }}
         {{- end }}
       labels:
-        {{- include "newrelic.selectorLabels" . | nindent 8 }}
+        {{- include "common.labels.selectorLabels" . | nindent 8 }}
+        {{- include "common.labels.podLabels" . | nindent 8 }}
         app.kubernetes.io/component: controlplane
-        {{- with .Values.podLabels }}
-          {{- toYaml . | nindent 8 }}
-        {{- end }}
-        mode: {{ include "newrelic.mode" . }}
     spec:
-      {{- with .Values.images.pullSecrets }}
+      {{- with include "common.images.renderPullSecrets" ( dict "pullSecrets" (list .Values.images.pullSecrets) "context" .) }}
       imagePullSecrets:
-          {{- toYaml . | nindent 8 }}
+        {{- . | nindent 8 }}
       {{- end }}
-      serviceAccountName: {{ template "newrelic.serviceAccountName" . }}-controlplane
-      {{- if include "newrelic.controlPlane.hostNetwork" . }}
+      {{- with include "common.dnsConfig" . }}
+      dnsConfig:
+        {{- . | nindent 8 }}
+      {{- end }}
+      {{- if ( include "common.hostNetwork" . ) | default .Values.controlPlane.hostNetwork }}
```

Comment:
The same I said [up here](https://github.com/newrelic/nri-kubernetes/pull/383#discussion_r845368587):

![Screenshot 2022-04-07 at 18 48 37](https://user-images.githubusercontent.com/53659978/162255570-7646a711-29d4-4516-92c8-6a40440d7017.png)

Virtually is the same as putting directly this:
```suggestion
{{- .Values.controlPlane.hostNetwork }}
```

Because of the way `default` threats falsiness... it is just to keep the precedence.

Reply:
You were completely right, I don't know what I was thinking when I wrote that (and it seems that I was completely convinced of that).

Solved on [#a1b6f77](https://github.com/newrelic/nri-kubernetes/pull/383/commits/a1b6f77ff8c7384c6873cd52c42aa9ca2068523c)